README.md

```diff
@@ -39,96 +46,184 @@ jobs:
               uses: hashicorp/vault-action@v2.3.1
               with:
                 url: https://vault.mycompany.com:8200
-                token: ${{ secrets.VaultToken }}
-                caCertificate: ${{ secrets.VAULTCA }}
+                token: ${{ secrets.VAULT_TOKEN }}
+                caCertificate: ${{ secrets.VAULT_CA_CERT }}
                 secrets: |
                     secret/data/ci/aws accessKey | AWS_ACCESS_KEY_ID ;
                     secret/data/ci/aws secretKey | AWS_SECRET_ACCESS_KEY ;
                     secret/data/ci npm_token
             # ...
 ```
 
-## Authentication method
+## Authentication Methods
 
-While most workflows will likely use a vault token, you can also use an `approle` to authenticate with Vault. You can configure which by using the `method` parameter:
+Consider using a [Vault authentication method](https://www.vaultproject.io/docs/auth) such as the JWT auth method with
+[GitHub OIDC tokens](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect) or the AppRole auth method. You can configure which by using the `method` parameter.
+
+### JWT with GitHub OIDC Tokens
+
+You can configure trust between a GitHub Actions workflow
+and Vault using the
+[GitHub's OIDC provider](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect).
+Each GitHub Actions workflow receives an auto-generated OIDC token with claims
+to establish the identity of the workflow.
+
+__Vault Configuration__
+
+<details>
+<summary>Click to toggle instructions for configuring Vault.</summary>
+
+Set up Vault with the [JWT auth method](https://www.vaultproject.io/api/auth/jwt#configure).
+Pass the following parameters to your auth method configuration:
+
+- `oidc_discovery_url`: `https://token.actions.githubusercontent.com`
+- `bound_issuer`: `https://token.actions.githubusercontent.com`
+
+
+Configure a [Vault role](https://www.vaultproject.io/api/auth/jwt#create-role) for the auth method.
+
+- `role_type`: `jwt`
+
+- `bound_audiences`: `["sigstore"]`. Update this parameter if you change
+  the `aud` claim in the GitHub OIDC token.
```

Comment:
It might be useful to mention here that the default `aud` is `http://github.com/<org>`?

```suggestion
- `bound_audiences`: `"http//github.com/<org>"`. Update this parameter if
  you change the `aud` claim in the GitHub OIDC token via the
  `jwtGithubAudience` parameter in the action config.
```

Also when testing this, I was reminded that even though `bound_audiences` is an array, if you use the CLI it's easier to specify multiple audiences by setting the parameter multiple times. i.e.

```
vault write auth/jwt/role/demo \
      bound_subject="repo:tvoran/test-github-action:ref:refs/heads/main" \
      bound_audiences="https://github.com/tvoran" \
      bound_audiences="sigstore" \
      ...
```

Otherwise I think you have to resort to json input to set an actual array? All that to say it might be good to remove the [] brackets from the example setting.